src/test/java/htsjdk/samtools/util/IoUtilTest.java

```diff
@@ -187,4 +191,35 @@ public void testFileTypeUnix(final String path, boolean expectedIsRegularFile) {
                 {"/non/existent/file", Boolean.TRUE},
         };
     }
+    
+    @Test
+    public void isGZIPInputStreamTest() throws IOException {
+        // test string without compression
+        final byte message[]="Hello World".getBytes();
+        BufferedInputStream in = new BufferedInputStream(new ByteArrayInputStream(message),10);
+        Assert.assertFalse(IOUtil.isGZIPInputStream(in));
+        // call twice to verify 'in.reset()' was called
+        Assert.assertFalse(IOUtil.isGZIPInputStream(in));
+        in.close();
+        
+        //compress the message
+        ByteArrayOutputStream bos =new ByteArrayOutputStream();
+        GZIPOutputStream gzout=new GZIPOutputStream(bos);
+        gzout.write(message);
+        gzout.finish();
+        gzout.close();
+        //and now test for compression
+        in = new BufferedInputStream(new ByteArrayInputStream(bos.toByteArray()),10);
+        Assert.assertTrue(IOUtil.isGZIPInputStream(in));
+        // call twice to verify 'in.reset()' was called
+        Assert.assertTrue(IOUtil.isGZIPInputStream(in));
+        in.close();
+    }
+    @DataProvider(name = "mayBeGZippedInputStreamTestCases")
+    private Object[][] mayBeGZippedInputStreamTestCases() {
+        return new Object[][]{
+                {"Hello World", Boolean.FALSE},

```

Comment:
you can just use the constants `false` and `true` instead `Boolean.FALSE`,
also no need to put the hello world in the dataprovider if it's always the same.